mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/RabbitMQEventBus.java

```diff
@@ -44,29 +45,45 @@
     private static final boolean DURABLE = true;
     private static final String DIRECT_EXCHANGE = "direct";
 
-    private final Sender sender;
-    private final GroupRegistrationHandler groupRegistrationHandler;
-    private final KeyRegistrationHandler keyRegistrationHandler;
-    private final EventDispatcher eventDispatcher;
+    private final Mono<Connection> connectionMono;
+    private final EventSerializer eventSerializer;
+    private final RoutingKeyConverter routingKeyConverter;
+
+    private GroupRegistrationHandler groupRegistrationHandler;
+    private KeyRegistrationHandler keyRegistrationHandler;
+    private EventDispatcher eventDispatcher;
+    private AtomicBoolean isRunning;
+    private Sender sender;
 
     RabbitMQEventBus(RabbitMQConnectionFactory rabbitMQConnectionFactory, EventSerializer eventSerializer, RoutingKeyConverter routingKeyConverter) {
-        Mono<Connection> connectionMono = Mono.fromSupplier(rabbitMQConnectionFactory::create).cache();
-        this.sender = RabbitFlux.createSender(new SenderOptions().connectionMono(connectionMono));
-        this.groupRegistrationHandler = new GroupRegistrationHandler(eventSerializer, sender, connectionMono);
-        this.eventDispatcher = new EventDispatcher(eventSerializer, sender);
-        this.keyRegistrationHandler = new KeyRegistrationHandler(eventSerializer, sender, connectionMono, routingKeyConverter);
+        this.connectionMono = Mono.fromSupplier(rabbitMQConnectionFactory::create).cache();
+        this.eventSerializer = eventSerializer;
+        this.routingKeyConverter = routingKeyConverter;
+
     }
 
     public void start() {
-        eventDispatcher.start();
-        keyRegistrationHandler.start();
+        if (isRunning == null || !isRunning.get()) {
+            sender = RabbitFlux.createSender(new SenderOptions().connectionMono(connectionMono));
+            groupRegistrationHandler = new GroupRegistrationHandler(eventSerializer, sender, connectionMono);
+            eventDispatcher = new EventDispatcher(eventSerializer, sender);
+            keyRegistrationHandler = new KeyRegistrationHandler(eventSerializer, sender, connectionMono, routingKeyConverter);
+
+            eventDispatcher.start();
+            keyRegistrationHandler.start();
+
+            isRunning = new AtomicBoolean(true);
```

Comment:
You defeat the purpose of the AtomicBoolean by reallocating it.

Prefer `isRunning.set(true);`